### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by kvnlay**

### CODE REVIEW

**Comment by talihomz** [comment]
## Good Job Overall!
---

Great work on this @kvnlay @mosaaleb! I've added some suggestions to your code below.

File: enumerable_methods.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      yield(object[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    object = self
+    object.my_each do |x|
+      (new_arr << x) if yield(x) == true
```
Comment (talihomz) [inline enumerable_methods.rb]: This conditional could be simplified by having it simply as `(new_arr << x) if yield(x)`

File: enumerable_methods.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    end
+    new_arr
+  end
+
+  def my_all?
+    object = self
+    object.my_each do |x|
+      return false if yield(x) == false
```
Comment (talihomz) [inline enumerable_methods.rb]: This conditional could be simplified by having it simply as `(new_arr << x) unless yield(x)`

File: enumerable_methods.rb [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    end
+    true
+  end
+
+  def my_any?
+    object = self
+    object.my_each do |x|
+      return true if yield(x) == true
```
Comment (talihomz) [inline enumerable_methods.rb]: This could be simplified into `return yield(x)`

File: enumerable_methods.rb [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    end
+    false
+  end
+
+  def my_none?
+    object = self
+    object.my_each do |x|
+      return false if yield(x) == true
```
Comment (talihomz) [inline enumerable_methods.rb]: This could be simplified into return `!yield(x)`

File: enumerable_methods.rb [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    true
+  end
+
+  def my_count
+    object = self
+    return object.size unless block_given?
+
+    count = 0
+    object.my_each do |x|
+      (count += 1) if yield(x) == true
```
Comment (talihomz) [inline enumerable_methods.rb:60]: This could be simplified into `(count += 1) if yield(x)`

**Review by talihomz — CHANGES_REQUESTED** [review]

File: enumerable_methods.rb (hunk 3 above)
Comment (mosaaleb) [reply enumerable_methods.rb]: Hi @talihomz 
`return yield x` will always return `false` unless only the first element match the condition, In this case, I need to loop through all the elements until the end. right?

File: enumerable_methods.rb (hunk 4 above)
Comment (mosaaleb) [reply enumerable_methods.rb]: Also @talihomz , `!yield x` will only return whatever the condition of the first element? and I need to iterate till the end.

**Comment by talihomz** [comment]
Great job with this!

👍

**Pull request merged**